cfpb/cfgov-refresh
# Fixing error notification on the Leadership PDF Calendar

Fixing error notification on the Leadership PDF Calendar

**This PR contains commits from https://github.com/cfpb/cfgov-refresh/pull/2250**

## Changes

- Updated `cfgov/v1/util/util.py` to make the error message dynamic through string interpolation. 
- Updated `cfgov/jinja2/v1/about-us/the-bureau/leadership-calendar/print/index.html` to fix issue with logic and formatting. 
- Updated `cfgov/cal/views.py` to add validation for the `CalendarPDFForm1`.
- Updated `cfgov/jinja2/v1/_includes/macros/filters.html` to fix issue rendering the notification.


## Testing

- Disable Javascript in the browser.
- Visit the `http://localhost:8000/about-us/the-bureau/leadership-calendar/` page. 
- Click the `Download PDF` button. You should see an error notification once redirected. 
- Enter in `123456` in the `From :` field. Click the `Download PDF` button. You should see an error notification once redirected. 

## Review

- @richaagarwal 
- @chosak 
- @kurtrwall 

## Screenshots

<img width="906" alt="screen shot 2016-07-29 at 5 50 56 pm" src="https://cloud.githubusercontent.com/assets/1696212/17264461/117c370c-55b5-11e6-902c-e7110127b830.png">


## Notes

- Most of the code was contained in https://github.com/cfpb/cfgov-refresh/blob/master/cfgov/v1/forms.py. It probably makes sense to delete that code at this time, since much of it has been migrated to views.py. 

- I am removing all query-string parameters when redirecting. It might make sense to add back the valid parameters. 

- There are two forms on the leadership-calendar page sharing the same query-string parameters. We might want to consider an alternative approach.

## Todos

- Write Tests. 

## Checklist
